camel/logger.py

```diff
@@ -0,0 +1,107 @@
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ========= Copyright 2023-2024 @ CAMEL-AI.org. All Rights Reserved. =========
+
+import logging
+import os
+import sys
+
+# Create a private logger
+_logger = logging.getLogger('camel')
+
+
+def _configure_library_logging():
+    if os.environ.get('CAMEL_LOGGING_DISABLED', 'False').lower() == 'true':
+        return
+
+    if not logging.root.handlers and not _logger.handlers:
+        logging.basicConfig(
+            level=os.environ.get('LOGLEVEL', 'INFO').upper(),
+            format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+            stream=sys.stdout,
+        )
+        logging.setLoggerClass(logging.Logger)
+        _logger.info("Camel library logging has been configured.")
+    else:
+        _logger.debug("Existing logger configuration found, using that.")
+
+
+def disable_logging():
+    r"""
+    Disable all logging for the Camel library.
+
+    This function sets the log level to a value higher than CRITICAL,
+    effectively disabling all log messages, and adds a NullHandler to
+    suppress any potential warnings about no handlers being found.
+    """
+    os.environ['CAMEL_LOGGING_DISABLED'] = 'true'
+    _logger.setLevel(logging.CRITICAL + 1)
+    _logger.addHandler(logging.NullHandler())
```

Comment:
`disable_logging` function adds a `NullHandler` each time it's called, leading to potential redundancy